app/src/main/java/net/gsantner/markor/format/FormatRegistry.java

```diff
@@ -101,6 +106,16 @@ public static FormatRegistry getFormat(int formatId, @NonNull final Context cont
         final AppSettings appSettings = ApplicationObject.settings();
 
         switch (formatId) {
+            case FORMAT_CSV: {
```

Comment:
General comment: We really need to spend some time making each format cleaner and more independent. One has to touch like 10 files to add a format and this can lead to bugs + reduced maintainability. Ideally we should just have to implement the classes and 'register' them in one place

Reply:
I agree. it was painfull for me to find all places i have to fix to make csv work. Probably i missed some.

this should be done once your "make permissions simpler" is finished.